src/DataTypes/DataTypeDecimalBase.h

```diff
@@ -72,7 +72,7 @@ class DataTypeDecimalBase : public DataTypeWithSimpleSerialization
     {
         if (unlikely(precision < 1 || precision > maxPrecision()))
             throw Exception("Precision " + std::to_string(precision) + " is out of bounds", ErrorCodes::ARGUMENT_OUT_OF_BOUND);
-        if (unlikely(scale < 0 || static_cast<UInt32>(scale) > maxPrecision()))
+        if (unlikely(scale > maxPrecision()))
```

Comment:
This was strange code from @4ertus2 
Also I'm surprised that tautological comparison is not detected by compiler.